### PR TITLE
Integrate WordPressKit and WordPressAuthenticator unit tests

### PR DESCRIPTION
**Pull request by crazytonyli**
> [!Note]
> This PR is built on top of https://github.com/wordpress-mobile/WordPress-iOS/pull/23366

This PR integrates the exiting unit tests of WordPressKit and WordPressAuthenticator into Xcode and CI.

From Xcode, you run their unit test from the Test navigator.

Screenshot 2024-06-18 at 1 50 03 PM

### CODE REVIEW

File: WordPressKit/WordPressKit.podspec [hunk 1]
```diff
@@ -30,4 +30,25 @@ Pod::Spec.new do |s|
...
+    test_spec.dependency 'OHHTTPStubs/Swift', '~> 9.0'
+    test_spec.dependency 'OCMock', '~> 3.4'
+    test_spec.dependency 'Alamofire', '~> 5.0'
+
+    test_spec.source_files = 'Tests/**/*.{h,m,swift}'
+    test_spec.resource_bundles = {
+      # The files in this bundle are duplicated at the root of the test bundle,
+      # because some tests are still looking for them at the root directory,
+      # instead of the `CoreAPITests` bundle.
+      'CoreAPITests' => 'Tests/CoreAPITests/Stubs/**/*',
```
Comment (dangermattic) [inline WordPressKit/WordPressKit.podspec:45]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wordpress-mobile/WordPress-iOS/blob/a82a815b1276904f94207c5eae5e2fc86654aeb6/WordPressKit/WordPressKit.podspec#L45"></span>Style/TrailingCommaInHashLiteral: Avoid comma after the last item of a hash.</td>
    </tr>
  </tbody>
</table>

File: WordPressKit/WordPressKit.podspec [hunk 2]
```diff
@@ -30,4 +30,25 @@ Pod::Spec.new do |s|
...
+    }
+    test_spec.resources = [
+      'Tests/WordPressKitTests/Mock Data/**/*',
+      'WordPressKitTests/**/*',
+      'Tests/CoreAPITests/Stubs/**/*',
+      'Tests/**/*.{json,html,xml}',
```
Comment (dangermattic) [inline WordPressKit/WordPressKit.podspec:51]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wordpress-mobile/WordPress-iOS/blob/a82a815b1276904f94207c5eae5e2fc86654aeb6/WordPressKit/WordPressKit.podspec#L51"></span>Style/TrailingCommaInArrayLiteral: Avoid comma after the last item of an array.</td>
    </tr>
  </tbody>
</table>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by kean** [comment]
- Is this change in the right order? The [next PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/23378) seems to be undo-ing a large part of it because instead of CocoaPods it uses Xcode targets for these two new modules.
- (enhancement) This change adds a large amount of tests to the project and two separate jobs that will have to run for every PR. The WPKit and WPAuthenticator tests themselves run relatively fast, but there is a non-trivial amount of time spent in `Setting up Gems` and `Setting up Pods`. I would suggest adding unit tests to the existing `wordpress-unit-tests` that already runs tests for multiple targets.
- (nit) I would name it `WordPressKitTests` instead of `WordPressKit-Unit-Tests` to be consistent with other test targets
- Just noting that WPKit has some flaky tests, adding to the existing flaky test suite. The more flaky tests, the more the chance one of them fails the build. I would suggest removing the flaky ones – no big loss.

**Comment by crazytonyli** [comment]
@kean The original plan is just moving WPKit and WPAuthenticator to this repo, which is completed once this PR is merged. But later, I thought I'll go ahead to try replacing CocoaPods pods with Swift Packages, since WPKit was the blocker from adopting Swift Package in our own libraries. That went pretty well, so I carried on and replaces as many CocoaPods pods as I can, which is done in https://github.com/wordpress-mobile/WordPress-iOS/pull/23378.

> This change adds a large amount of tests to the project and two separate jobs that will have to run for every PR. 

This CI step is gone in https://github.com/wordpress-mobile/WordPress-iOS/pull/23378. They now run alongside with WordPress unit tests, as you suggested.

> I would name it WordPressKitTests instead of WordPressKit-Unit-Tests

This is also done in https://github.com/wordpress-mobile/WordPress-iOS/pull/23378. BTW, the name "Xxx-Unit-Tests" was created by CocoaPods. We don't have much control over it.

> Just noting that WPKit has some flaky tests

I'll try rerun WPKit unit tests locally (and possibly on CI too), to see if I can fix those flaky tests. Removing them can be the last resort if we can't fix them properly.

**Comment by crazytonyli** [comment]
@kean Can we get this PR reviewed and merged first? I'll follow up with a separate PR if unit tests require some fixes.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/292">25.2</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by kean** [comment]
> @kean Can we get this PR reviewed and merged first? I'll follow up with a separate PR if unit tests require some fixes.

Sure, I'm not suggesting to redo anything – was just clarifying.

> This CI step is gone in https://github.com/wordpress-mobile/WordPress-iOS/pull/23378. They now run alongside with WordPress unit tests

Nice!

**Review by kean — APPROVED** [review]
LGTM
*as long as the tests completes and assuming that most of it is reworked in the next PR. The source changes to the tests look good.

**Comment by crazytonyli** [comment]
```
[2024-06-20T22:12:50Z] Errors:
[2024-06-20T22:12:50Z] - [ ] WordPressAuthenticator/WordPressAuthenticator.podspec#L1: Naming/FileName: The name of this source file (`WordPressAuthenticator.podspec`) should use snake_case.
[2024-06-20T22:12:50Z] - [ ] WordPressAuthenticator/WordPressAuthenticator.podspec#L3: Metrics/BlockLength: Block has too many lines. [38/25]
[2024-06-20T22:12:50Z] - [ ] WordPressKit/WordPressKit.podspec#L1: Naming/FileName: The name of this source file (`WordPressKit.podspec`) should use snake_case.
[2024-06-20T22:12:50Z] - [ ] WordPressKit/WordPressKit.podspec#L3: Metrics/BlockLength: Block has too many lines. [35/25]
```

I'm going to ignore the above issues reported by danger, because these podspec files will be deleted soon.

**Pull request merged**